datalad/crawler/pipelines/balsa.py

```diff
@@ -0,0 +1,219 @@
+                self.repo.remove(item)
+                files = list(f_f('.*', topdir='_files'))
+            else:
+                lgr.warning("%s does not exist in the individaully listed files by name, "
+                            "but will be kept from canconical tarball" % item)
+        if files:
+            lgr.warning("The following files do not exist in the canonical tarball, but are "
+                        "individually listed files and will not be kept: %s" % files)
+            rmtree(files_path)
+
+        yield data
+
+
+def pipeline(dataset_id, url=TOPURL):
+    lgr.info("Creating a pipeline for the BALSA dataset %s" % dataset_id)
+    annex = Annexificator(create=False,
+                          statusdb='json',
+                          special_remotes=[ARCHIVES_SPECIAL_REMOTE, DATALAD_SPECIAL_REMOTE],
+                          options=["-c",
+                                   "annex.largefiles="
+                                   "exclude=Makefile and exclude=LICENSE* and exclude=ISSUES*"
+                                   " and exclude=CHANGES* and exclude=README*"
+                                   " and exclude=*.[mc] and exclude=dataset*.json"
+                                   " and exclude=*.txt"
+                                   " and exclude=*.json"
+                                   " and exclude=*.tsv"
+                                   ])
+
+    if not exists("_files"):
+        makedirs("_files")
+
+    def splitpath(data):
+        data = data.copy()
+        fullpath = data.pop('path')
+        path = os.path.dirname(fullpath)
+        if path:
+            data['path'] = path
+        data['filename'] = os.path.basename(fullpath)
+        yield data
+
+    files_url = opj(url, 'file/show/')  # files_url = https://balsa.wustl.edu/file/show/
+
+    url = opj(url, 'study/')  # url = https://balsa.wustl.edu/study/
+    dataset_url = '%sshow/%s' % (url, dataset_id)  # url = https://balsa.wustl.edu/study/show/[dataset_id]
+
+    balsa = BalsaSupport(repo=annex.repo)
+
+    # BALSA has no versioning atm, so no changelog either
+    return [
+        annex.switch_branch('incoming'),
+        [
+            crawl_url(url),
+            [
+                assign({'dataset': dataset_id}),
+                skip_if({'dataset': 'test study upload'}, re=True),
+            ],
+            [
+                crawl_url(dataset_url),
+                [
+                    extract_meta,
+                    annex,
+                ],
+                [
+                    # canonical tarball
+                    a_href_match('.*/download/.*', min_count=1),
+                    annex,
+                ],
+                [
+                    a_href_match(files_url, min_count=2),
+                    assign({'path': '_files/%(url_text)s'}, interpolate=True),
+                    sub({'path': {' / ': '/'}}),
+                    splitpath,
+                    annex,
```

Comment:
dataset_url is for the 'show' page to allow the fetching of: canonical tarball ( done by finding the actual 'download' link that exists solely on the show page), the individually listed files, and txt info to create descriptor file -> essentially everything is on this show page. For the canonical tarball, it is the actual file from the download link (in that sub pipeline). 
In what context did you not see the warning? Files are verified by context in the verify_files (look at line 117/118) with get_file_key. I'll create a specific test for this as well.
Have you run the pipeline for RVVG? I have not yet due to the sizeable download involved- only tested with WG33 which does not have this scenario. I'll test.